src/application.cpp

```diff
@@ -355,6 +359,15 @@ void Application::showEdgeColorDialog()
     emit actionTriggered(StateMachine::Action::EdgeColorChanged);
 }
 
+void Application::showGridColorDialog()
+{
+    const auto color = QColorDialog::getColor(Qt::white, m_mainWindow.get());
+    if (color.isValid()) {
+        m_mediator->setGridColor(color);
+    }
+    emit actionTriggered(StateMachine::Action::GridColorChanged);
```

Comment:
Should this emit be inside the validation if..?

Reply:
Yes, but i did it cause other color functions have same logic. For examle see 2 functions above (showEdgeColorDialog and showBackgroundColorDialog)
Maybe make this change in all functions?